icepyx/core/query.py

```diff
@@ -753,20 +754,18 @@ def granules(self):
 
         Examples
         --------
-        >>> reg_a = ipx.Query('ATL06',[-55, 68, -48, 71],['2019-02-20','2019-02-28']) # doctest: +SKIP
+        >>> reg_a = ipx.Query('ATL06',[-55, 68, -48,
+                                       71],['2019-02-20','2019-02-28']) #
+        +SKIP
         >>> reg_a.granules # doctest: +SKIP
         <icepyx.core.granules.Granules at [location]>
         """
-
-        if not hasattr(self, "_granules") or self._granules is None:
-            self._granules = Granules()
-
-        return self._granules
```

Comment:
Looks to me like `self.granules` is static and could be set at init-time as a class attribute, but then we wouldn't get this awesome docstring. I changed the method to a `cached_property` to, IMO, get the best qualities of both.